Arcade Validatesdk self build is not restoring the correct versions during the build process

@dotnet-mc-bot commented on [Wed Jan 09 2019](https://github.com/dotnet/core-eng/issues/4945)

There were a set of failures during this build. Here is a summary of these:
* https://dnceng.visualstudio.com/internal/_build/results?buildId=69051
  - **Agent:** a0000J0
  - **Error log:** ##[error]PowerShell exited with code '1'....

@jcagme, @markwilkie


---

@robertborr commented on [Wed Jan 09 2019](https://github.com/dotnet/core-eng/issues/4945#issuecomment-452814343)

@alexperovich , @riarenas  - are working on this as I was researching it.

--- a/eng/validate-sdk.ps1
+++ b/eng/validate-sdk.ps1
@@ -88,8 +88,9 @@ try {
   StopDotnetIfRunning
   
   Write-Host "Building with updated dependencies"
-
-  . .\common\build.ps1 -configuration $configuration @Args  /p:AdditionalRestoreSources=$packagesSource
+  # Clear this global variable so that we don't use the cached toolset that we restored in Step 1.
+  Remove-Variable _ToolsetBuildProj -Scope Global
+  & .\common\build.ps1 -configuration $configuration @Args  /p:AdditionalRestoreSources=$packagesSource
 }
 catch {
   Write-Host $_

